std/traits.d

```diff
@@ -6727,14 +6727,29 @@ unittest
  * This is not recursive; it will not search for symbols within symbols such as
  * nested structs or unions.
  */
-template getSymbolsByUDA(alias symbol, alias attribute)
-{
-    import std.typetuple : Filter, staticMap, TypeTuple;
+template getSymbolsByUDA(alias symbol, alias attribute) {
+    import std.string : format;
+    import std.meta : AliasSeq, Filter;
+
+    // translate a list of strings into symbols. mixing in the entire alias
+    // avoids trying to access the symbol, which could cause a privacy violation
+    template toSymbols(names...) {
+        static if (names.length == 1)
+            mixin("alias toSymbols = AliasSeq!(symbol.%s);".format(names[0]));
+        else
+            mixin("alias toSymbols = AliasSeq!(symbol.%s, toSymbols!(names[1..$]));"
+                  .format(names[0]));
```

Comment:
> Could we somehow replace this with __traits(getMember, Type, name)

Only if we change the API to only retrieve publically accessible members (even if those members are visible in the caller's module)

> Or maybe we have to change the API to return a bunch of strings like __traits(allMembers, Type)

I don't think this matters as we'd still be using a mixin for `hasSpecificUDA`. Get rid of that mixin, and we can only support members visible to `std.traits`.

I'm not completely sure I understand the problem here, can you elaborate or give an example?